File: 필수문항/123_연속구간.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            String num = br.readLine();
            if (num == null) {
                return;
            }

            Character before = num.charAt(0);
            int ans = 1;
            int now = 1;
            for (int i = 1; i < num.length(); i++) {
                if (before == num.charAt(i)) {
                    now++;
                    ans = Math.max(ans, now);
                } else {
                    ans = Math.max(ans, now);
                    before = num.charAt(i);
                    now = 1;
                }
            }
            System.out.println(ans);
        }

    }
}

File: 필수문항/123_연속구간.py
import sys

input = sys.stdin.readline

while True:
    num = input().rstrip()
    if num == "":
        break

    before = num[0]
    ans = 1
    now = 1
    for i in range(1, len(num)):
        if num[i] == before:
            now += 1
            ans = max(ans, now)
        else:
            before = num[i]
            ans = max(ans, now)
            now = 1
    print(ans)
